src/xdsl/irdl_mlir_printer.py

```diff
@@ -0,0 +1,125 @@
+from io import IOBase
+from xdsl.printer import Printer
+from xdsl.dialects.irdl import (
+    DialectOp, ParametersOp, TypeOp, OperandsOp, ResultsOp, OperationOp,
+    EqTypeConstraintAttr, AnyTypeConstraintAttr, AnyOfTypeConstraintAttr,
+    DynTypeBaseConstraintAttr, DynTypeParamsConstraintAttr,
+    TypeParamsConstraintAttr, NamedTypeConstraintAttr)
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True, eq=False)
+class IRDLPrinter:
+
+    stream: IOBase
+
+    def _print(self, s: str, **kw_args):
+        print(s, file=self.stream, **kw_args)
+
+    def print_module(self, module):
+        self._print('module{')
+        module.walk(lambda di: IRDLPrinter.print_dialect_definition(self, di)
+                    if isinstance(di, DialectOp) else None)
+        self._print('}')
+
+    def print_type_definition(self, type: TypeOp):
+        self._print(f"    {type.name} {type.attributes['type_name'].data} {{")
```

Comment:
```suggestion
        self._print(f"    {TypeOp.name} {type.attributes['type_name'].data} {{")
```
This is to make clear what is happening, and also will save you from bugs if `type` has the wrong type.